src/fshtypes/Extension.ts

```diff
@@ -19,4 +21,13 @@ export class Extension extends FshEntity {
     this.mixins = [];
     this.rules = [];
   }
+
+  ruleIsAllowed(rule: Rule) {
+    return !(
+      rule instanceof InsertRule ||
+      rule instanceof MappingRule ||
+      rule instanceof ValueSetComponentRule ||
+      rule instanceof ConceptRule
+    );
+  }
```

Comment:
Although the list of disallowed `Rule` types is shorter, I would prefer for this to be implemented directly as checking a list of allowed types. With an allow-list implementation, new `Rule` types are automatically permitted, which seems more error prone than requiring the new type to be explicitly allowed.

Reply:
I think I agree with @mint-thompson -- let's be explicit about what we allow.  I also think something like below might read more easily:
```ts
function ruleIsAllowed(rule: Rule) {
  switch (rule.constructor) {
    case InsertRule:
    case CardRule:
    case FlagRule:
      return true;
    default:
      return false;
  }
}
```
There is a limitation -- it wouldn't work for _subclasses_ of these things (while `instanceof` would) -- but I don't think that's really a problem for us.  We don't have subclasses of these, and even if we did, maybe we should explicitly test for them anyway.

Reply:
OR... we could just have a single file where we manage the allowed and disallowed rules.  I tested that out and confirmed something like this would also work:

```ts
// This map is obviously incomplete, but proves the point...
const allowedRulesMap = new Map<any, any[]>([
  [Profile, [InsertRule, CardRule, FlagRule]],
  [FshValueSet, [InsertRule, ValueSetComponentRule]]
]);

export function isAllowedRule(fshDefinition: Profile | FshValueSet, rule: Rule): boolean {
  return allowedRulesMap.get(fshDefinition.constructor)?.find(r => r === rule.constructor) != null;
}
```

Of course the reverse map (where the rule is the key and entities are the values) would also work.  I'd have to type it all out to see which one would be preferred though.

Reply:
I'll try to do something like the map and be explicit. We do have subtypes though on `ValueSetComponentRule`, but there are only 2 subtypes so I will just list them as well.

Reply:
I didn't test it, but if we want to support subtypes with the map, then something like this _might_ work:
```ts
export function isAllowedRule(fshDefinition: Profile | FshValueSet, rule: Rule): boolean {
  return allowedRulesMap.get(fshDefinition.constructor)?.find(r => rule instanceof r) != null;
}
```
(Darn -- deleted all my test code for this about 20 minutes ago!)